src/nupic/research/connections.py

```diff
@@ -26,111 +26,87 @@
                   # other floats
 
 
+
 class Segment(object):
   """ Class containing minimal information to identify a unique segment """
 
-  __slots__ = ['idx', 'cell', 'data']
+  __slots__ = ["cell", "idx", "flatIdx", "_synapses", "_numDestroyedSynapses",
+               "_destroyed", "_lastUsedIteration"]
 
-  def __init__(self, idx, cell, data):
+  def __init__(self, cell, idx, flatIdx):
     """
-       @param idx  (int) Index of the segment on the cell
-       @param cell (int) Index of the cell that this segment is on
+    @param cell (int) Index of the cell that this segment is on.
+    @param idx (int) Index of the segment on the cell.
+    @param flatIdx (int) The segment's flattened list index.
     """
-    self.idx = idx
     self.cell = cell
-    self.data = data
+    self.idx = idx
+    self.flatIdx = flatIdx
+    self._synapses = []
+    self._numDestroyedSynapses = 0
+    self._destroyed = False
+    self._lastUsedIteration = -1
 
 
   def __eq__(self, other):
-    return (self.idx, self.cell) == (other.idx, other.cell)
+    """ Explicitly implement this for unit testing. The flatIdx is not designed
+    to be consistent after serialize / deserialize.
+
+    """
+    return ((self.idx, self.cell, self._synapses, self._numDestroyedSynapses,
```

Comment:
Personally, I find this difficult to read.

I prefer the following, even if it's more verbose:

```python
return (self.idx == other.idx and
        self.cell == other.cell and
        self._synapses == other._synapses and
        self._numDestroyedSynapses == other._numDestroyedSynapses and
        self._destroyed == other._destroyed and
        self._lastUsedIteration == other._lastUsedIterator)
```

or:

```python
return all((self.idx == other.idx,
            self.cell == other.cell,
            self._synapses == other._synapses,
            self._numDestroyedSynapses == other._numDestroyedSynapses,
            self._destroyed == other._destroyed,
            self._lastUsedIteration == other._lastUsedIterator))
```